src/bucket.js

```diff
@@ -80,6 +80,7 @@ Bucket.prototype.createPutUrl = function(prefix, options) {
  */
 Bucket.prototype.createGetUrl = function(prefix, forceS3 = false) {
   assert(prefix, 'prefix must be given');
+  prefix = encodeURIComponent(prefix);
```

Comment:
I'm surprised it's worked this long without this... Really this should be a separate PR... and we should be on hand to monitor what happens when you land this...

Well, at least, a separate PR so that it's easy to back out, and then ping sherifs to look out for weird artifact errors while you deploy. You have my **r+** on this change. It seems like you did the due diligence on testing things.

Reply:
it's worked so long because the vast majority of cases where there'd be a URI Encoding substitution don't cause an issue with S3 (e.g. / chars).  It's weird how S3 accepts a partial URI encoding.

A second PR doesn't improve the deployment or backout options really.  This is already its own commit, so the backout would be `git revert 227d7aebd0563ec6fd0ccc1ced5d4f85fc7dbbb6`.

Reply:
I think we need to only encode the minimum number of characters... So the URLs are still pretty...

Things like `/` that don't need encoding should not be encoded...  I suspect there is code that relies on URLs being pretty.

Because of curl or wget that probably does things like parsing the last slash to determine the file name they are writing to...